frontend/packages/noobaa-storage-plugin/src/components/object-bucket-claim-page/modals/delete-confirm-modal.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {
+  createModalLauncher,
+  ModalBody,
+  ModalComponentProps,
+  ModalTitle,
+  ModalSubmitFooter,
+} from '@console/internal/components/factory/modal';
+import { K8sKind, k8sKill } from '@console/internal/module/k8s';
+import {
+  HandlePromiseProps,
+  history,
+  resourceListPathFromModel,
+  withHandlePromise,
+} from '@console/internal/components/utils';
+import { getName, getNamespace, YellowExclamationTriangleIcon } from '@console/shared';
+import { TextInput } from '@patternfly/react-core';
+
+const ConfirmDeleteModal = withHandlePromise((props: ConfirmDeleteModalProps) => {
+  const [inputMatch, setInputMatch] = React.useState(false);
+
+  const { close, cancel, handlePromise, inProgress, errorMessage, redirectTo } = props;
+
+  const submit = (event) => {
+    event.preventDefault();
+    const { kind, resource } = props;
+    const promise = handlePromise(k8sKill(kind, resource, {}));
+    promise
+      .then(() => {
+        const re = new RegExp(`/${getName(resource)}(/|$)`);
+        if (re.test(window.location.pathname)) {
```

Comment:
Instead of getting the path name from window.location, can you try fetching the value from props.match?  It should be available there as well.